spec/models/miq_worker/container_common_spec.rb

```diff
@@ -96,21 +96,29 @@ def deployment_name_for(name)
       [
         {:subject => MiqGenericWorker.new, :name => deployment_name_for("generic")},
         {:subject => MiqUiWorker.new,      :name => deployment_name_for("ui")},
-        {:subject => ManageIQ::Providers::Openshift::ContainerManager::EventCatcher.new(:queue_name => "ems_2"), :name => deployment_name_for("openshift-container-event-catcher-2")},
-        {:subject => ManageIQ::Providers::Redhat::InfraManager::MetricsCollectorWorker.new, :name => deployment_name_for("redhat-infra-metrics-collector")}
+        {:subject => ManageIQ::Providers::Openshift::ContainerManager::EventCatcher.new(:queue_name => "ems_2"), :name => deployment_name_for("openshift-container-events-2")},
+        {:subject => ManageIQ::Providers::Redhat::InfraManager::MetricsCollectorWorker.new, :name => deployment_name_for("redhat-infra-metrics")}
       ]
     end
 
     it "returns the correct name for each worker" do
       test_cases.each { |test| expect(test[:subject].worker_deployment_name).to eq(test[:name]) }
     end
 
+    # bundle exec rspec spec/models/miq_worker/container_common_spec.rb:111
     it "no worker deployment names are over 60 characters", :providers_common => true do
       # OpenShift does not allow deployment names over 63 characters
       # We also want to leave some for the ems_id so we compare against 60 to be safe
-      MiqWorkerType.seed
-      MiqWorkerType.pluck(:worker_type).each do |klass|
-        expect(klass.constantize.new.worker_deployment_name.length).to be <= 60
+      MiqWorkerType.seed 
+      MiqWorkerType.pluck(:worker_type).sort.each do |klass|
+        worker = klass.constantize.new
+        name = worker.worker_deployment_name(true)
+
+        # easier to read, but doesn't work in github issues
+        #puts "| #{worker.worker_deployment_name(false).size} => #{worker.worker_deployment_name(true).size} | #{worker.shortened_class_name_before} | #{worker.shortened_class_name_after} | #{name} |"
+        puts "| #{worker.worker_deployment_name(false).size} => #{worker.worker_deployment_name(true).size} | #{worker.worker_deployment_name(false)} | #{worker.worker_deployment_name(true)} |"
+
+        #expect(name.length).to be <= 60
```

Comment:
This needs to stay (and yes, I realize this is WIP), but I'm wondering if the 3 we gave as wiggle room for the ems_id is enough.  Since in your example you had 8 chars `35r12345`, which seems reasonable, perhaps this should be `expect(name.length).to be <= 55`

Reply:
I took this out so we could see all the names.
If I leave it in mine, then it stops after only a hand full of names.

Hmm... now that I have shortened them enough, maybe it is good enough now